staging/src/k8s.io/api/policy/v1/types.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// PodDisruptionBudgetSpec is a description of a PodDisruptionBudget.
+type PodDisruptionBudgetSpec struct {
+	// An eviction is allowed if at least "minAvailable" pods selected by
+	// "selector" will still be available after the eviction, i.e. even in the
+	// absence of the evicted pod.  So for example you can prevent all voluntary
+	// evictions by specifying "100%".
+	// +optional
+	MinAvailable *intstr.IntOrString `json:"minAvailable,omitempty" protobuf:"bytes,1,opt,name=minAvailable"`
+
+	// Label query over pods whose evictions are managed by the disruption
+	// budget.
```

Comment:
I don't see the selector change proposed in https://github.com/kubernetes/enhancements/blob/master/keps/sig-apps/85-Graduate-PDB-to-Stable/README.md#fix-handling-of-empty-selector-in-disruption-controller implemented here

Reply:
Yeah, I haven't done this yet. Just wanted to get the basics in place first. This is next on the list.

Reply:
I have updated the PR with an attempt to address this, but I'm not sure it is correct. So my assumptions are:
 * We can't change the version of PDB used in the controller yet, since we want to update the API first and then only update the controller in the next release.
 * There will existing PDBs stored in etcd, so we also can't change the semantics of an empty selector in the internal type.

The result is that the conversion must happen in the v1 api as we convert between v1 and the internal type.

Reply:
> There will existing PDBs stored in etcd, so we also can't change the semantics of an empty selector in the internal type.

etcd stores in v1beta1, not internal. I would expect to make adjustments in the `v1beta1 <-> internal` conversion

Reply:
> We can't change the version of PDB used in the controller yet, since we want to update the API first and then only update the controller in the next release.

if you want to add a conformance test, the controller must use the v1 API (conformance tests must pass in clusters with no alpha/beta features or APIs enabled)

Reply:
That makes sense. I have updated the PR to handle conversion in v1beta1 and updated the controller to use the v1 api.